lib/src/clojure_lsp/feature/file_management.clj

```diff
@@ -80,12 +80,15 @@
         usage-signature (juxt :to :name)]
     (find-changed-elems-by usage-signature old-var-usages new-var-usages)))
 
-(defn reference-filenames [filename old-local-analysis new-local-analysis db]
-  (let [changed-var-definitions (find-changed-var-definitions old-local-analysis new-local-analysis)
+(defn reference-filenames [filename db-before db-after]
+  (let [old-local-analysis (get-in db-before [:analysis filename])
+        new-local-analysis (get-in db-after [:analysis filename])
+        changed-var-definitions (find-changed-var-definitions old-local-analysis new-local-analysis)
         changed-var-usages (find-changed-var-usages old-local-analysis new-local-analysis)
-        project-analysis (into {}
-                               q/filter-project-analysis-xf
-                               (dissoc (:analysis db) filename)) ;; don't notify self
+        project-db (-> db-after
+                       q/db-with-project-analysis
```

Comment:
the need to use this function outside query.clj maybe tells that we should move the query used on this function to inside query, WDYT?